`make clean / distclean` breaks the build

Following instructions in the NuPIC README, I executed the following sequence

```
mkdir -p $NUPIC/build/scripts
cd $NUPIC/build/scripts
cmake $NUPIC
make clean / distclean
make -j3
```

The pertinent environment variables are:

```
NTAX_DEVELOPER_BUILD is NOT defined

$ echo $NUPIC
/Users/vkruglikov/nta/nupic
$ echo $NTA
/Users/vkruglikov/nta/nupic/build/release
$ echo $NTA_ROOTDIR 
/Users/vkruglikov/nta/nupic/build/release
$ echo $PYTHONPATH 
/Users/vkruglikov/Packages/Cython-0.15.1:/Users/vkruglikov/nta/nupic-darwin64/lib/python2.7/site-packages::/Users/vkruglikov/nta/nupic/build/release/lib/python2.7/site-packages:/Users/vkruglikov/nta/nupic/build/release/lib/python2.7/site-packages
```

The failure looks like this:

```
vkruglikov@vkruglikovs-mbp:~/nta/nupic/build/scripts (master)$ make -j3
[  6%] [ 12%] [ 18%] Building CXX object CMakeFiles/support.dir/lang/py/support/PyArray.cpp.o
Building CXX object CMakeFiles/support.dir/lang/py/support/NumpyVector.cpp.o
Building CXX object CMakeFiles/support.dir/lang/py/support/PyHelpers.cpp.o
In file included from /Users/vkruglikov/nta/nupic/lang/py/support/PyHelpers.cpp:24:
/Users/vkruglikov/nta/nupic/lang/py/support/PyHelpers.hpp:34:10: fatal error: 'nta/utils/Log.hpp' file not found
#include <nta/utils/Log.hpp>
         ^
In file included from /Users/vkruglikov/nta/nupic/lang/py/support/PyArray.cpp:30:
In file included from /Users/vkruglikov/nta/nupic/lang/py/support/PyArray.hpp:39:
/Users/vkruglikov/nta/nupic/lang/py/support/PyHelpers.hpp:34:10: fatal error: 'nta/utils/Log.hpp' file not found
#include <nta/utils/Log.hpp>
         ^
In file included from /Users/vkruglikov/nta/nupic/lang/py/support/NumpyVector.cpp:33:
In file included from /Users/vkruglikov/nta/nupic-darwin64/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:15:
In file included from /Users/vkruglikov/nta/nupic-darwin64/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:17:
In file included from /Users/vkruglikov/nta/nupic-darwin64/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1728:
/Users/vkruglikov/nta/nupic-darwin64/lib/python2.7/site-packages/numpy/core/include/numpy/npy_deprecated_api.h:11:2: warning: "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API
      NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by #defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION"
 ^
In file included from /Users/vkruglikov/nta/nupic/lang/py/support/NumpyVector.cpp:45:
/Users/vkruglikov/nta/nupic/lang/py/support/NumpyVector.hpp:32:10: fatal error: 'nta/types/types.hpp' file not found
#include <nta/types/types.hpp> // For nta::Real.
         ^
1 warning and 1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/support.dir/lang/py/support/NumpyVector.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/support.dir/lang/py/support/PyHelpers.cpp.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/support.dir/lang/py/support/PyArray.cpp.o] Error 1
make[1]: *** [CMakeFiles/support.dir/all] Error 2
make: *** [all] Error 2
```

If I skip `make clean / distclean`, then the build completes successfully.

--- a/README.md
+++ b/README.md
@@ -69,7 +69,6 @@ Set the following environment variables in your `~/.bashrc` file. `$NUPIC` is th
 #### Build:
 
     cd $NUPIC/build/scripts
-    (optional) make clean / distclean   # to compile all the files again
     make -j3
 
 > **Note**: -j3 option specify '3' as the maximum number of parallel jobs/threads that Make will use during the build in order to gain speed. However, you can increase this number depending your CPU.

